model/host/host.go

```diff
@@ -1413,10 +1413,19 @@ func FindHostsToTerminate() ([]Host, error) {
 			{ // hosts that failed to provision
 				StatusKey: evergreen.HostProvisionFailed,
 			},
-			{ // hosts that are taking too long to provision
-				"$or": []bson.M{
-					bson.M{ProvisionedKey: false},
-					bson.M{StatusKey: evergreen.HostProvisioning},
+			{ // hosts that are either taking too long to provision or have started tasks before provisioning finished but not communicated recently
+				"$and": []bson.M{
+					{"$or": []bson.M{
+						{ProvisionedKey: false},
+						{StatusKey: evergreen.HostProvisioning},
+					}},
+					{
+						"$or": []bson.M{
+							{RunningTaskKey: bson.M{"$exists": false}},
+							{LTCTaskKey: ""},
```

Comment:
This check shouldn't be there should it? I think this will cause the query to match hosts that aren't provisioned but finished running at least 1 task, and may be running a second one

Reply:
The reason I made this ticket is because user data hosts that don't get to the running status but nonetheless are running agents and tasks should not be terminated unless they go silent for some reason (e.g. externally terminated). If we don't include this check, hosts that are not in the "running" state will terminate after 25 minutes, even if they're running a task just fine. mms encountered this problem yesterday when cloud host ready jobs weren't running but user data hosts were running their tasks, then being terminated by the 25 minute provisioning cutoff.

I tried to make what it's doing more explicit.